Older versions of OS X require including some more headers before using
if_nametoindex, as documented in its man page.

--- src/core/lib/iomgr/grpc_if_nametoindex_posix.cc.orig	2019-07-03 04:44:40.000000000 +1000
+++ src/core/lib/iomgr/grpc_if_nametoindex_posix.cc	2019-11-18 07:47:39.000000000 +1100
@@ -25,6 +25,8 @@
 #include "src/core/lib/iomgr/grpc_if_nametoindex.h"
 
 #include <errno.h>
+#include <sys/types.h>
+#include <sys/socket.h>
 #include <net/if.h>
 
 #include <grpc/support/log.h>
